core/cloudflow-operator/src/main/scala/cloudflow/operator/Name.scala

```diff
@@ -110,6 +110,9 @@ object Name {
   def ofConfigMap(streamletDeploymentName: String) =
     makeDNS1123Compatible(s"configmap-${fixDots(streamletDeploymentName)}")
 
+  def ofLabelValue(name: String) =
```

Comment:
Should this not be truncate to 63 chars?

Reply:
https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/ @yuchaoran2011

Reply:
Looks like label names (not just values) are also limited to 63.

Reply:
https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture/identifiers.md#identifiers-and-names-in-kubernetes

`rfc1035/rfc1123 label (DNS_LABEL): An alphanumeric (a-z, and 0-9) string, with a maximum length of 63 characters, with the '-' character allowed anywhere except the first or last character, suitable for use as a hostname or segment in a domain name.`

Reply:
Yeah I saw that label name has that restriction too (hence the `ofLabelValue` method name). But I didn't do it because I checked that all our label names are hard coded (e.g. `Operator.StreamletNameLabel` or `Operator.AppIdLabel `) and therefore at no risk of running over the limit. Do you think we need to truncate those too to preempt those default names changing to a longer version some day?

Reply:
We could do a type that converts to string that breaks if it goes over the 63.
Something simple like (not-compiled example):
```
case class LabelValue(label:String){
  require(label.size <=63, "label has max length of 63")
  override def toString() = label
}
```
and then use that in definitions

Reply:
And very important, this comment refers to line 114 using the wrong function @yuchaoran2011 .

Reply:
Now changed to just doing the truncation.

Reply:
@RayRoestenburg Is this good to be merged now?